src/common/cbsd.py

```diff
@@ -0,0 +1,934 @@
+#!/usr/bin/env python
+# -- encoding: utf-8 --
+
+# Project: CBSD to SAS interface
+__author__ = "Francisco Benavides (francisco.benavides@federatedwireless.com)"
+"""
+    CBSD Message Response
+    Federated Wireless
+
+    SAS to User Protocol Technical Specification
+    WINNF-16-S-0016 SAS to CBSD v0.3.10, 12 July 2016
+
+    9.2     Message Transport
+            HTTP v1.1, Content-type: application/json
+            JSON objects shall use UTF-8 enconding
+            All requests from CBSD to SAS shall use the HTTP POST method
+
+    10      Parameters for SAS-CBSD messages
+"""
+
+# Imports
+import unittest
+import logging
+from datetime import datetime
+import pytz
+
+# Constants
+
+# Log format
+LOG_FORMAT = logging.Formatter(
+    '%(asctime)-15s - CBSD - '
+    '%(funcName)s - %(levelname)s - %(message)s')
+# MeasCapability allowed values
+MEAS_CAPABILITY_ALLOWED_VALUES = (
+    'EutraCarrierRssiNonTx',
+    'EutraCarrierRssiAlways')
+CHANNEL_TYPE_ALLOWED_VALUES = (
+    'PAL',
+    'GAA')
+PARAMETER_RESPONSE = (
+    'response',
+    'responseCode',
+    'responseData',
+    'responseMessage')
+PARAMETER_FREQUENCYRANGE = (
+    'lowFrequency',
+    'highFrequency')
+PARAMETER_OPERATIONPARAM = (
+    'operationParam',
+    'maxEirp',
+    'operationFrequencyRange') + PARAMETER_FREQUENCYRANGE
+MESSAGE_PARAMETERS_REGISTRATION = (
+    'registrationResponse',
+    'cbsdId',
+    'measReportConfig') + PARAMETER_RESPONSE
+MESSAGE_PARAMETERS_DEREGISTRATION = (
+    'deregistrationResponse',
+    'cbsdId') + PARAMETER_RESPONSE
+MESSAGE_PARAMETERS_SPECTRUMINQUIRY = (
+    'spectrumInquiryResponse',
+    'cbsdId',
+    'availableChannel',
+    'frequencyRange',
+    'channelType',
+    'ruleApplied') + PARAMETER_RESPONSE + PARAMETER_FREQUENCYRANGE
+MESSAGE_PARAMETERS_GRANT = (
+    'grantResponse',
+    'cbsdId',
+    'grantId',
+    'grantExpireTime',
+    'heartbeatInterval',
+    'measReportConfig',
+    'channelType') + PARAMETER_RESPONSE + PARAMETER_OPERATIONPARAM
+MESSAGE_PARAMETERS_HEARTBEAT = (
+    'heartbeatResponse',
+    'cbsdId',
+    'grantId',
+    'transmitExpireTime',
+    'grantExpireTime',
+    'heartbeatInterval',
+    'measReportConfig') + PARAMETER_RESPONSE + PARAMETER_OPERATIONPARAM
+MESSAGE_PARAMETERS_RELINQUISHMENT = (
+    'relinquishmentResponse',
+    'cbsdId',
+    'grantId') + PARAMETER_RESPONSE
+
+# Classes
+
+
+class CBSD(object):
```

Comment:
Answering your question: it is not enough to only check that the correct value comes in, we need to verify that other parameters that should also accompany or should not come at all, are present or not and thus comply with the protocol spec exactly as it is written, no more nor less.
To also complement the previous answer: there is only one place where we know we verify each message, and follow the protocol as to: verify that we receive the required parameters, and the values and types are correct, then the same for the conditional parameters and at the end the optional parameters, if any.
We also make sure, no additional parameters arrive, and if they do so we raise a warning.
Also, if per each combination of expected responseCode value we were to check all the parameter combinatios as per the spec, and so and so.... it would make each test case both to large and messy and prone to error.
And if you look inside the code, you would see we do use the assertEqual that comes with unittest, as well as various other unittest provided assertions, accordingly.